src/test/java/org/opensearch/knn/index/KNNESSettingsTestIT.java

```diff
@@ -106,7 +106,11 @@ public void testCreateIndexWithInvalidSpaceType() throws IOException {
     }
 
     public void testUpdateIndexSetting() throws IOException {
-        Settings settings = Settings.builder().put("index.knn", true).put(KNNSettings.KNN_ALGO_PARAM_EF_SEARCH, 512).build();
+        Settings settings = Settings.builder()
+            .put("index.knn", true)
+            .put(KNNSettings.KNN_ALGO_PARAM_EF_SEARCH, 512)
+            .put(KNNSettings.INDEX_KNN_ADVANCED_APPROXIMATE_THRESHOLD, 0)
```

Comment:
for cases like this I think we should just move the code to use `getDefaultIndexSettings` so that in future we don't need to make the changes.